src/Voron/Impl/Paging/Pager.cs

```diff
@@ -332,6 +334,7 @@ public void EnsureContinuous(ref State state, long requestedPageNumber, int numb
         }
         Debug.Assert(totalAllocatedSize >= state.TotalAllocatedSize, "totalAllocatedSize >= state.TotalAllocatedSize");
         state = new State(this, readAddress, writeAddress, totalAllocatedSize, handle);
+        NativeMemory.RegisterFileMapping(state.Pager.FileName, new IntPtr(state.ReadAddress), state.TotalAllocatedSize, null);
```

Comment:
I wonder whether we shouldn't do it inside `State` constructor. Then both calls:
- `NativeMemory.RegisterFileMapping()`
- `NativeMemory.UnregisterFileMapping()`

would be in the same file. Which actually makes sense since register / unregister should always be called on creation / disposal of the pager state.

Reply:
Good idea